app/controllers/articleController.js

```diff
@@ -138,4 +139,100 @@ export default class ArticleController {
       }
     }
   }
+  /**
+   * search Method
+   *
+   * This method searches and returns all articles matching the given params
+   *
+   * @param {HTTP REQ} req web request object
+   * @param {HTTP RES} res web response object
+   * @param {function} next middleware function
+   */
+  static async search(req, res){
+    async function getUserIdsByEmailsQuery(listOfEmails) {
+      if(!Array.isArray(listOfEmails)) {
+        throw new ErrorWrongQueryType();
+      }
+      // returns ids of the user objects given as a list of emails
+      const allUsers = [];
+        for (let i = 0; i < listOfEmails.length; i++) {
+          const user = await UsersAccessor.getUserByEmail(listOfEmails[i]);
+          allUsers[i] = user._id;
+        }
+        return {"$in": allUsers};
+    }
+
+    async function equalNumber(num) {
+      if (!Number.isNaN((Number(num)))) {
+        return num;
+      }
+      else {
+        throw new ErrorWrongQueryType();
+      }
+    }
+
+    async function equalString(str) {
+      if(typeof(str) === "string") {
+        return str;
+      }
+      else {
+        throw new ErrorWrongQueryType();
+      }
+    }
+
+    async function inQuery(cats) {
+      if(Array.isArray(cats)) {
+        return {"$in": cats};
+      }
+      else {
+        throw new ErrorWrongQueryType();
+      }
+    }
```

Comment:
No async here needed

Reply:
We will be able to remove these once we have incoming validation for the req.body with JSON Schema